go/kbfs/data/path_part_string.go

```diff
@@ -48,21 +52,48 @@ func NewPathPartString(
 	if len(ext) > extRestoreMaxLen && ext != tarGzSuffix {
 		ext = ""
 	}
-	conflictedIndex := strings.LastIndex(plaintext, ".conflicted (")
+
+	if strings.HasPrefix(plaintext, prefixFileInfo) {
+		toObfuscate = strings.TrimPrefix(plaintext, prefixFileInfo)
+	}
+
+	conflictedIndex := strings.LastIndex(plaintext, suffixConflictStart)
 	if conflictedIndex > 0 {
+		// If this is a conflict file, we should obfuscate everything
+		// besides the conflict part, to get a string that matches the
+		// non-conflict portion.  (Also continue to ignore any file
+		// info prefix when obfuscating, as above.)
+		if len(toObfuscate) == 0 {
+			toObfuscate = plaintext
+		}
+		toObfuscate = toObfuscate[:strings.LastIndex(
+			toObfuscate, suffixConflictStart)] + ext
 		ext = plaintext[conflictedIndex:]
 	}
-	return PathPartString{plaintext, ext, obfuscator}
+	return PathPartString{plaintext, toObfuscate, ext, obfuscator}
 }
 
 func (pps PathPartString) String() string {
-	if strings.HasPrefix(pps.plaintext, prefixToSkipObfsucation) ||
-		pps.obfuscator == nil {
+	if pps.obfuscator == nil {
+		return pps.plaintext
+	}
+
+	var prefix string
+	if strings.HasPrefix(pps.plaintext, prefixFileInfo) {
```

Comment:
I think it's confusing to take off the prefix and store the rest up on line 69, but to seemingly discard it and then check for it again down here. Unless you think it's important to save the space the prefix would take up, I'd advocate for keeping the prefix in the `PathPartString`.

Reply:
Yeah checking for the prefix again was partly for saving that space, but also it's somewhat required to check for it again explicitly since we have to distinguish it from the regular `.kbfs_` case. But i will make that into a `fileInfo` bool instead, and set the obfuscator to `nil` when it's a regular `kbfs_` prefix, to save some CPU on each call to `String()`.